packages/block-editor/src/components/segmented-text-control/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { BaseControl, Button } from '@wordpress/components';
+
+/**
+ * Control to facilitate segmented text selections.
+ *
+ * @param {Object}   props           Component props.
+ * @param {string}   props.label     A label for the control.
+ * @param {string}   props.value     Currently selected value.
+ * @param {Function} props.onChange  Callback to handle onChange.
+ * @param {Array}    props.controls  Array of controls to display.
+ * @param {string}   props.className Additional class name to apply.
+ *
+ * @return {Element} Element to render.
+ */
+export default function SegmentedTextControl( {
+	label,
+	value,
+	controls,
+	onChange,
+	className,
+} ) {
+	return (
+		<fieldset
```

Comment:
Should this use `BaseControl` so that it supports props like `help`? No preference here, just thinking out loud.

Reply:
I was under the impression the `BaseControl` wraps everything in a styled `div` component without passing through additional props so we'd lose the `fieldset`. My memory is failing me but I assume it was required for a11y.

If it was being changed, we'd probably need to remove the `as="legend"` from the `VisualLabel` as well. Maybe such a change would be best left to a follow-up?

Reply:
I don't know when such an implementation was first done, but I also think i11y is related. This kind of implementation (fieldset > legend + contents) is done in many components, so I think this is fine for now.